File: nj2k/testData/newJ2k/postProcessing/GetOperator.java
import java.util.HashMap;
import kotlinApi.KotlinClass;
import javaApi.JavaClass;

class X {
    int get(int index) {
        return 0;
    }
}

class C {
    String foo(HashMap<String, String> map) {
        return map.get("a");
    }

    int foo(X x) {
        return x.get(0);
    }

    int foo(KotlinClass kotlinClass) {
        return kotlinClass.get(0); // not operator!
    }

    int foo(JavaClass javaClass) {
        return javaClass.get(0);
    }
}

File: kotlin-ide/j2k/old/testData/fileOrElement/postProcessing/GetOperator.kt
// ERROR: Type mismatch: inferred type is String? but String was expected
import java.util.HashMap
import kotlinApi.KotlinClass
import javaApi.JavaClass

internal class X {
    operator fun get(index: Int): Int {
        return 0
    }
}

internal class C {
    fun foo(map: HashMap<String, String>): String {
        return map["a"]
    }

    fun foo(x: X): Int {
        return x[0]
    }

    fun foo(kotlinClass: KotlinClass): Int {
        return kotlinClass.get(0) // not operator!
    }

    fun foo(javaClass: JavaClass): Int {
        return javaClass.get(0)
    }
}